core/src/main/java/org/infinispan/remoting/transport/jgroups/CommandAwareRpcDispatcher.java

```diff
@@ -268,53 +247,45 @@ private void reply(org.jgroups.blocks.Response response, Object retVal, Replicab
       }
    }
 
-   protected static Message constructMessage(Buffer buf, Address recipient,ResponseMode mode, boolean rsvp,
-                                             DeliverOrder deliverOrder) {
-      Message msg = new Message();
-      msg.setBuffer(buf);
-      encodeDeliverMode(msg, deliverOrder);
+   static RequestOptions constructRequestOptions(ResponseMode mode, boolean rsvp, DeliverOrder deliverOrder,
+                                                 long timeout) {
+      RequestOptions options = new RequestOptions(mode, timeout);
+      encodeDeliverMode(options, deliverOrder);
       //some issues with the new bundler. put back the DONT_BUNDLE flag.
       if (deliverOrder == DeliverOrder.NONE || mode != ResponseMode.GET_NONE) {
-         msg.setFlag(Message.Flag.DONT_BUNDLE.value());
+         options.setFlags(Message.Flag.DONT_BUNDLE.value());
```

Comment:
should we remove the ```DONT_BUNDLE``` flag? it shown better performance without it during the F2F meeting.